conjureplugin/config/internal/v1/config.go

```diff
@@ -82,10 +88,118 @@ func (cfg *IRLocatorConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
+// This is the conversion logic used by UpgradeConfig.
+func (v1cfg *ConjurePluginConfig) ToV2() v2.ConjurePluginConfig {
+	// Create v2 config with intelligent field mapping
+	v2cfg := v2.ConjurePluginConfig{
+		ConfigWithVersion: versionedconfig.ConfigWithVersion{Version: "2"},
+		GroupID:           v1cfg.GroupID,
+		ProjectConfigs:    make(map[string]v2.SingleConjureConfig),
+	}
+
+	// Convert each project using the per-project conversion logic
+	for projectName, v1proj := range v1cfg.ProjectConfigs {
+		v2cfg.ProjectConfigs[projectName] = v1proj.ToV2(projectName)
+	}
+
+	// Check if we need to allow conflicting output directories.
+	// We detect conflicts the same way ToParams does: exact same directory AND parent-child relationships.
+	// Calculate the actual output directories for each project
+	outputDirs := make(map[string][]string)
+	for projectName, proj := range v2cfg.ProjectConfigs {
+		resolvedOutputDir := proj.ResolvedOutputDir(projectName)
+		outputDirs[resolvedOutputDir] = append(outputDirs[resolvedOutputDir], projectName)
+	}
+
+	v2cfg.AllowConflictingOutputDirs = len(GetConflictingOutputDirs(outputDirs)) > 0
+
+	return v2cfg
+}
+
 func UpgradeConfig(cfgBytes []byte) ([]byte, error) {
 	var cfg ConjurePluginConfig
 	if err := yaml.UnmarshalStrict(cfgBytes, &cfg); err != nil {
 		return nil, errors.Wrapf(err, "failed to unmarshal conjure-plugin v1 configuration")
 	}
+
+	cfgBytes, err := yaml.Marshal(cfg.ToV2())
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to marshal upgraded v2 configuration")
+	}
+
 	return cfgBytes, nil
 }
+
+func GetConflictingOutputDirs(outputDirToProjects map[string][]string) []error {
+	var warnings []error
+
+	sortedOutputDir := slices.Sorted(maps.Keys(outputDirToProjects))
+	for _, outputDir := range sortedOutputDir {
+		projects := outputDirToProjects[outputDir]
+		if len(projects) <= 1 {
+			continue
+		}
+		warnings = append(warnings, errors.Errorf("Projects %v are configured with the same outputDir %q, which may cause conflicts when generating Conjure output", projects, outputDir))
+	}
+
+	for i, dir1 := range sortedOutputDir {
+		for _, dir2 := range sortedOutputDir[i+1:] {
+			if isChild(dir1, dir2) || isChild(dir2, dir1) {
+				projects1 := outputDirToProjects[dir1]
+				projects2 := outputDirToProjects[dir2]
+				warnings = append(warnings, errors.Errorf("Projects %v (outputDir %q) and %v (outputDir %q) have a parent-child directory relationship, which may cause conflicts when generating Conjure output", projects1, dir1, projects2, dir2))
+			}
+		}
+	}
+
+	return warnings
+}
+
+// isChild checks if child is a subdirectory of parent.
+// Paths are normalized with filepath.Clean before comparison.
+func isChild(parent, child string) bool {
+	parent = filepath.Clean(parent)
+	child = filepath.Clean(child)
+	rel, err := filepath.Rel(parent, child)
+	return err == nil && !strings.HasPrefix(rel, "..") && rel != "."
+}
```

Comment:
This logic isn't unique to `v1` config -- for clarify, move to a directory/package like `conjureplugin/config/internal/validate/validate.go`